earthfile2llb/earthfile2llb.go

```diff
@@ -162,6 +168,17 @@ func Earthfile2LLB(ctx context.Context, target domain.Target, opt ConvertOpt, in
 			}
 			opt.ForceSaveImage = true // legacy mode always saves images regardless of locally or remotely referenced
 		}
+		if opt.waitBlock != nil {
+			panic("waitBlock should not be set during initial call")
+		}
+		opt.waitBlock = newWaitBlock()
+
+		defer func() {
+			err := opt.waitBlock.wait(ctx) // ensure final waitblock completes
+			if retErr == nil {
+				retErr = err
+			}
+		}()
```

Comment:
Feels like we could decouple the boolean `initialCall` from `opt.waitBlock` by simply checking `if opt.waitBlock == nil { ... do initialization logic ... }`.

(Similar to`if opt.SolveCache == nil`, `if opt.Visited == nil`, `if opt.MetaResolver == nil`, `if opt.ErrorGroup == nil` above)